### PR TITLE
MINOR: improve description of `commit.interval.ms` config

### PR DESCRIPTION
**Pull request by mjsax**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by photoszzt — COMMENTED** [review]
image
This change looks good to me. I think this LOW rating also needs to change. For at-least-once processing, it's indeed LOW importance for performance. But for exactly-once processing, it's in fact overwrite the linger.ms if commit.interval.ms is smaller than linger.ms and commit more frequently does impact the performance. I think in this page, people don't realize this if they don't understand what the underlying protocol is doing.

**Review by showuon — APPROVED** [review]
LGTM! Thanks.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -452,7 +452,9 @@ public class StreamsConfig extends AbstractConfig {
     /** {@code commit.interval.ms} */
     @SuppressWarnings("WeakerAccess")
     public static final String COMMIT_INTERVAL_MS_CONFIG = "commit.interval.ms";
-    private static final String COMMIT_INTERVAL_MS_DOC = "The frequency in milliseconds with which to save the position of the processor." +
+    private static final String COMMIT_INTERVAL_MS_DOC = "The frequency in milliseconds with which to commit processing progress." +
+        " For at-least-once processing, committing means to save the position (ie, offsets) of the processor." +
+        " For exactly-once processing, it means to commit the transaction which includes to save the position." +
```
Comment (cadonna) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: Should we also mention that the interval with which transactions are committed also affects the latency downstream?

**Review by Kvicii — APPROVED** [review]

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java (hunk 1 above)
Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: Not sure -- would we go down the rabbit hole?

**Comment by mjsax** [comment]
I am not sure if bumping the priority from LOW to MEDIUM is the right thing to do? In general, we cannot also not explain all config cross-dependencies (ie, `linger.ms`). If there is no objection, I would merge the PR as-is?

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java (hunk 1 above)
Comment (cadonna) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: When I read the updated doc, the first question that comes to my mind is, what "commit the transaction" actually means. I would not like to leave a user with that question in mind. You could add something like 

```suggestion
        " For exactly-once processing, it means to commit the transaction which includes to save the position and to make the committed data in the output topic visible to consumers with isolation level read_committed." +
```

There are more to transactions that just that, but I think those are the main points.

Mine is just a suggestion, if you think it is not needed, I am fine.

**Review by cadonna — APPROVED** [review]
LGTM!

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java (hunk 1 above)
Comment (ableegoldman) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: +1 to what Bruno said -- the current wording, `"to save the position"` is more like an implementation detail (sort of) but we should be explaining this in terms of what it actually affects for the user/what's visible to them, which is as Bruno describes to make the data actually readable

**Pull request merged**